velociraptor/catalogue/registration.py

```diff
@@ -1022,44 +1022,135 @@ def registration_gas_hydrogen_species_masses(
+                "LogIronOverHydrogen": "Fe_over_H",
             }[long_species]
             element_name = {
-                "OxygenOverHydrogenMasses": "Oxygen",
-                "IronOverHydrogenMasses": "Iron",
+                "LogOxygenOverHydrogen": "Oxygen",
+                "LogIronOverHydrogen": "Iron",
             }[long_species]
             fraction_name = {
-                "OxygenOverHydrogenMasses": "(O/H)",
-                "IronOverHydrogenMasses": "(Fe/H)",
+                "LogOxygenOverHydrogen": "O/H",
+                "LogIronOverHydrogen": "Fe/H",
             }[long_species]
+
+            short_floortype = {"Lo": "lowfloor", "Hi": "highfloor",}[floor_type]
+            floor_value = {"Lo": "-4", "Hi": "-3",}[floor_type]
         except KeyError:
             raise RegistrationDoesNotMatchError
 
-        full_name = f"{element_name} Abundance Weighted Gas Mass {fraction_name}$\times M_{{\\rm gas}}$ ({aperture_size} kpc)"
-        snake_case = f"{short_species}_times_gas_mass_{aperture_size}_kpc"
+        full_name = f"""Log10 {element_name} Abundance Weighted {part_type.capitalize()} Mass ({fraction_name}) 
+        $\times M_{{\\rm gas}}$, from particle values floored at [{fraction_name}]$\\gtreq {floor_value}$ 
+        ({aperture_size} kpc)"""
+        snake_case = f"log_{short_species}_times_{part_type}_mass_{short_floortype}_{aperture_size}_kpc"
+        return unit, full_name, snake_case
+    else:
+        raise RegistrationDoesNotMatchError
+
+
+def registration_lin_element_ratios_times_masses(
+    field_path: str, unit_system: VelociraptorUnits
+) -> (unyt.Unit, str, str):
+    """
+    Registers the Fe/H times mass and O/H times mass within apertures for two particle floor values
+    """
+
+    unit = unit_system.mass
+
+    # Capture aperture size
+    match_string = "Aperture_([a-zA-Z]*)Masses_aperture_total_([a-zA-Z]*)_([0-9]*)_kpc"
+    regex = cached_regex(match_string)
+
+    match = regex.match(field_path)
+
+    if match:
+        long_species = match.group(1)
+        part_type = match.group(2)
+        aperture_size = match.group(3)
+
+        try:
+            short_species = {
+                "TotalOxygenOverHydrogen": "O_over_H_total",
+                "OxygenOverHydrogen": "O_over_H",
+                "IronOverHydrogen": "Fe_over_H",
+            }[long_species]
+            element_name = {
+                "TotalOxygenOverHydrogen": "Oxygen",
+                "OxygenOverHydrogen": "Oxygen",
+                "IronOverHydrogen": "Iron",
+            }[long_species]
+            fraction_name = {
+                "TotalOxygenOverHydrogen": "O/H",
+                "OxygenOverHydrogen": "O/H",
+                "IronOverHydrogen": "Fe/H",
+            }[long_species]
+        except KeyError:
+            raise RegistrationDoesNotMatchError
 
+        full_name = f"""Linear {element_name} Abundance Weighted {part_type.capitalize()} Mass ({fraction_name}) 
```

Comment:
Don't do strings like this; you end up with a `\n` and loads of spaces in the string. You should do:

```
full_name = (
  f"Linear {element_name} Abundance Weighted {part_type.capitalize()} "
  f"Mass ({fraction_name}) $\\times M_{{\\rm gas}}$ ({aperture_size} kpc)"
```

Additionally, it's good practice to _always_ escape backslashes.